src/pages/LoginPage.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  Text,
+  VStack,
+  useToast,
+  useColorModeValue,
+  InputGroup,
+  InputRightElement,
+} from '@chakra-ui/react';
+import { ViewIcon, ViewOffIcon } from '@chakra-ui/icons';
+import { useNavigate, Link, useLocation } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from '../store/store';
+import { loginUser } from '../features/user/userSlice';
+import CustomToast from '../components/CustomToast';
+
+const LoginPage: React.FC = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const toast = useToast();
+  const dispatch = useDispatch<AppDispatch>();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
```

Comment:
It is better to put all these states and their functions in a separate hook to shorten this component, and leave all the logic in the hook